src/util/records.py

```diff
@@ -106,6 +106,36 @@ def remove_view(cr, xml_id=None, view_id=None, silent=False, key=None):
         for [v_id] in cr.fetchall():
             remove_view(cr, view_id=v_id, silent=silent, key=xml_id)
 
+    # Occurrences of xml_id in the t-call of views are to be found and removed.
+    if xml_id != "?":
```

Comment:
> You can incorporate this conditional into the one existing above.

Yes, I can, but in odoo <= 8.0, the condition checking for the existence of the key column will fail because the key column doesn't exist. Therefore, my script won't run in such case.

Reply:
We could do this:
```
if xml_id != "?":
    if column_exists(cr, "ir_ui_view", "key"):
        cr.execute("SELECT id FROM ir_ui_view WHERE key = %s AND id != %s", [xml_id, view_id])
        for [v_id] in cr.fetchall():
            remove_view(cr, view_id=v_id, silent=silent, key=xml_id)
    ...
    >your part<
```

Reply:
I prefer the former version, it touches less code :/